src/app/betting/feature/accept-bets-page.component.ts

```diff
@@ -0,0 +1,176 @@
+import { TableModule } from 'primeng/table';
+import { NgOptimizedImage } from '@angular/common';
+import { ButtonModule } from 'primeng/button';
+import { RouterLink } from '@angular/router';
+import {
+  BetStatus,
+  showMessageOnError,
+} from '../../shared/util/supabase-helpers';
+import { StronglyTypedTableRowDirective } from '../../shared/ui/strongly-typed-table-row.directive';
+import { BetService } from '../../shared/data-access/bet.service';
+import { MessageService } from 'primeng/api';
+
+@Component({
+  selector: 'joshies-place-bet-choose-player-page',
+  standalone: true,
+  imports: [
+    PageHeaderComponent,
+    HeaderLinkComponent,
+    TableModule,
+    NgOptimizedImage,
+    ButtonModule,
+    RouterLink,
+    StronglyTypedTableRowDirective,
+  ],
+  template: `
+    <joshies-page-header headerText="Accept Bets" alwaysSmall>
+      <joshies-header-link
+        text="GM Tools"
+        routerLink=".."
+        chevronDirection="left"
+      />
+    </joshies-page-header>
+
+    @if (displayBets(); as bets) {
+      <p-table
+        [value]="bets"
+        [defaultSortOrder]="-1"
+        sortField="score"
+        [sortOrder]="-1"
+        [scrollable]="true"
+      >
+        <ng-template pTemplate="header">
+          <tr>
+            <th style="width: 60%;"></th>
+            <th></th>
+          </tr>
+        </ng-template>
+        <ng-template
+          pTemplate="body"
+          [joshiesStronglyTypedTableRow]="bets"
+          let-bet
+        >
+          <tr>
+            <!-- Bet Terms -->
+            <td>
+              <div class="flex flex-column gap-2 -py-2">
+                <div>
+                  {{ bet.requesterName }} wagers:
+                  {{ bet.requesterWager }}
+                </div>
+                <div>
+                  {{ userNameAndScore() }} wagers:
+                  {{ bet.yourWager }}
+                </div>
+                <div>{{ bet.description }}</div>
+              </div>
+            </td>
+            <!-- Accept/Reject Buttons -->
+            <td>
+              <div
+                class="text-right flex gap-2 flex-column md:flex-row justify-content-end"
+              >
+                <p-button
+                  label="Accept Bet"
+                  icon="pi pi-check"
+                  styleClass="w-full"
+                  (onClick)="acceptBet(bet.id)"
+                  [disabled]="
+                    bet.yourWager > (userPlayer()?.score ?? 0) ||
+                    bet.requesterWager > (bet.requesterScore ?? 0)
+                  "
+                  [loading]="submitting()"
+                />
+                <p-button
+                  label="Reject Bet"
+                  icon="pi pi-times"
+                  styleClass="w-full"
+                  (onClick)="rejectBet(bet.id)"
+                  [loading]="submitting()"
+                />
+              </div>
+            </td>
+          </tr>
+        </ng-template>
+      </p-table>
+    }
```

Comment:
We should probably add a message for when there are no pending bets, so it's not a blank screen. Maybe something like this:

```html
} @else {
  <p class="text-500 font-italic text-center mt-5">
    No pending bets
  </p>
}
```

Reply:
for some reason we weren't hitting the @else (I suspect something to do with the assignment to bets?). I resolved it by first checking displayBets()?.length, but let me know if you have a better idea

Reply:
When there's no pending bets, `displayBets()` returns an empty array (`[]`), which is still "truthy". So we do need to check `@if (displayBets()?.lenght)`.

But the full `@if` statement doesn't seem to be working as expected. We might need 2 nested `@if` statements: 1 to check `displayBets()?.length`, and 1 to assign the `displayBets` template variable.

This seems to work:

```html
@if (displayBets()?.length) {
  @if (displayBets(); as displayBets) {
    <!-- ... -->
  }
} @else {
  <p>No pending bets</p>
}
```